src/index.ts

```diff
@@ -22,37 +52,60 @@ export const goSync = <T, E extends Error>(fn: () => T): GoResult<T, E> => {
   }
 };
 
-const createGoError = <E extends Error>(err: unknown): GoResultError<E> => {
-  if (err instanceof Error) return fail(err);
-  return fail(new Error('' + err));
-};
+export const go = async <T, E extends Error>(
+  fn: Promise<T> | (() => Promise<T>),
+  options?: PromiseOptions
+): Promise<GoResult<T, E>> => {
+  const attemptOptions: AttemptOptions<any> = {
+    delay: options?.retryDelayMs || 200,
+    maxAttempts: options?.retries || 1,
```

Comment:
One more point question about `maxAttempts`:
>The maximum number of attempts or 0 if there is no limit on number of attempts.

In Airnode this was defined as: `maxAttempts: options.retries + 1`

The default for the library is 3, but do we also want to support `0`/unlimited or the way it is in Airnode currently?

Reply:
> In Airnode this was defined as: maxAttempts: options.retries + 1

Right. We're using retries so the number of attempts must be 1 (the original attempt) + the number of retries. I think that's fine.

> The default for the library is 3, but do we also want to support 0/unlimited or the way it is in Airnode currently?

What do you mean by "the way it is in Airnode currently"? Do we have unlimited retries somewhere?
But I think we should have as a default for the main `go` function 0 retries, which means no retries, just the one attempt and a default of 3 retries (1 attempt + 3 retries) for the `retryGo` and `retryTimeoutGo` functions.

Reply:
>Right. We're using retries so the number of attempts must be 1 (the original attempt) + the number of retries. I think that's fine.

Makes perfect sense, thanks!

> Do we have unlimited retries somewhere?
But I think we should have as a default for the main go function 0 retries, which means no retries, just the one attempt and a default of 3 retries (1 attempt + 3 retries) for the retryGo and retryTimeoutGo functions.

No unlimited retries are used currently, just wanted to confirm they're not needed :+1: